base/src/main/scala/clazz/Applicative.scala

```diff
@@ -2,8 +2,16 @@ package scato
 package clazz
 
 abstract class Applicative[F[_]] {
-  def apply: Apply[F]
+
   def pure[A](a: A): F[A]
+  def ap[A, B](fa: F[A])(f: F[A => B]): F[B]
+
+  def map[A, B](ma: F[A])(f: (A) => B): F[B] = ap(ma)(pure(f))
+
+  final val apply: Apply[F] = new Apply[F] {
```

Comment:
I see, so if I understand correctly:
```Bar``` has an instance of ```Apply``` and because ```Foo``` is a newtype which runtime representation is ```Bar```, you get the ```Apply[Foo]``` instance "for free" (via a safe-coerce somewhere?). It is then natural that we want to reuse ```Apply[Foo]``` to implements ```Applicative[Foo]```. right?

then, with this patch you would need to write (with ```fooApply: Apply[Foo]``` in scope) :

```scala
new Applicative[Foo]() {
  def pure[A](a: A): F[A] = ??? //TODO
  def ap[A, B](fa: F[A])(f: F[A => B]): F[B]  = fooApply.ap(fa)(f)

  def map[A, B](ma: F[A])(f: (A) => B): F[B] = fooApply.map(ma)(f)
}
```

Instead we would want the simpler following definition.

```scala
new Applicative[Foo]() {

  def pure[A](a: A): F[A] = ??? //TODO

  val apply = fooApply
}
```

In that case, your original definition is probably better. And the facilities implemented by this patch should in fact be one of the template traits you talk about in https://github.com/scalaz/scalaz/issues/1084 and not done directly in the main typeclasses.